crates/nu-engine/src/eval.rs

```diff
@@ -548,22 +548,36 @@ pub fn eval_expression(
             }
             Ok(Value::list(output, expr.span))
         }
-        Expr::Record(fields) => {
+        Expr::Record(items) => {
             let mut record = Record::new();
 
-            for (col, val) in fields {
-                // avoid duplicate cols.
-                let col_name = eval_expression(engine_state, stack, col)?.as_string()?;
-                let pos = record.index_of(&col_name);
-                match pos {
-                    Some(index) => {
-                        return Err(ShellError::ColumnDefinedTwice {
-                            second_use: col.span,
-                            first_use: fields[index].0.span,
-                        })
+            for item in items {
+                match item {
+                    RecordItem::Pair(col, val) => {
+                        // avoid duplicate cols.
+                        let col_name = eval_expression(engine_state, stack, col)?.as_string()?;
+                        let pos = record.index_of(&col_name);
+                        match pos {
+                            Some(index) => {
+                                return Err(ShellError::ColumnDefinedTwice {
+                                    second_use: col.span,
+                                    first_use: todo!(), // fields[index].0.span,
```

Comment:
This is going to require some thinking. I assume stuff like `{ a: 1, ...{ a: 2 } }` and `{ ...{ a: 2 }, a: 1 }` should be allowed, so we can't check `record.index_of`. Maybe all of the column names (from the `RecordItem::Pair`s) could be evaluated and checked before the values are evaluated?